### PR TITLE
Ask more questions during registration

### PR DESCRIPTION
**Pull request by KshitizSareen**
This is what the final result looks like:

Screenshot 2023-05-06 at 10 38 53 AM

I made changes to User.cs, the initial migration code of the database, the register.component.html, register.component.ts, account.service.ts, AccountController.cs and AccountRegisterVM.cs.

Please review it.

Regards,
Kshitiz.

### CODE REVIEW

File: src/Analysim.Web/ClientApp/src/app/register/register.component.ts [hunk 1]
```diff
@@ -48,24 +60,43 @@ export class RegisterComponent implements OnInit {
...
+    })
+
   }
 
 
   onSubmit() {
     // Variable for FormGroupValue
     let userReg = this.insertForm.value
 
+    let registrationSurvey = userReg.workPlace+";"+userReg.positionTitle+";"+userReg.whereDidYouHearAboutAnalysim+";"+userReg.wouldYouLikeToHearUpdatesFromUs.toString()+";"+userReg.registrationCode;
```
Comment (cengique) [inline src/Analysim.Web/ClientApp/src/app/register/register.component.ts]: @KshitizSareen the only correction I will ask is to change this string to be proper JSON, such as:
```
' { "workPlace": ' + userReg.workPlace + ', ...'
```
This way we can parse it in the database as well.

File: src/Analysim.Web/Controllers/AccountController.cs [hunk 2]
```diff
@@ -228,16 +231,39 @@ public async Task<IActionResult> Register([FromForm] AccountRegisterVM formdata)
             // Hold Error List
             List<string> errorList = new List<string>();
 
+            string registrationSurvey = formdata.RegistrationSurvey;
```
Comment (cengique) [inline src/Analysim.Web/Controllers/AccountController.cs:235]: You can convert this from/to JSON.

**Review by cengique — CHANGES_REQUESTED** [review]

File: src/Analysim.Web/Controllers/AccountController.cs (hunk 2 above)
Comment (KshitizSareen) [reply src/Analysim.Web/Controllers/AccountController.cs:235]: Hi @cengique.

I could not find any way to add json field in the migration file. I decided to leave registrationSurvey as a string field in the User class. I converted the string to json format like this:

let registrationSurvey = {"workPlace": userReg.workPlace,"positionTitle":userReg.positionTitle,"whereDidYouHearAboutAnalysim": userReg.whereDidYouHearAboutAnalysim,"wouldYouLikeToHearUpdatesFromUs": userReg.wouldYouLikeToHearUpdatesFromUs,"registrationCode":userReg.registrationCode};

I stringified it before sending to the backend.

 this.accountService.register(userReg.username, userReg.password, userReg.emailAddress,JSON.stringify(registrationSurvey))

Comment (cengique) [reply src/Analysim.Web/Controllers/AccountController.cs:235]: @KshitizSareen I think this works!

**Review by cengique — APPROVED** [review]
Looks very good!

**Pull request merged**